src/test/java/org/folio/rest/api/OaiPmhViewTest.java

```diff
@@ -385,6 +402,74 @@ public void testResponseStatus400WhenRequestingWithInvalidUntilDate()
     });
   }
 
+  @Test
+  public void test500(TestContext testContext) {
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(mock(HttpServerResponse.class));
+    AbstractInstanceRecordsAPIImpl.fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response.getStatus(), is(500));
+        }));
+  }
+
+  @Test
+  public void testTcpClose(TestContext testContext) {
+    HttpServerResponse httpServerResponse = mock(HttpServerResponse.class);
+    when(httpServerResponse.headWritten()).thenReturn(true);
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(httpServerResponse);
+    AbstractInstanceRecordsAPIImpl.fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response, is(nullValue()));
+          verify(httpServerResponse).close();
+        }));
+  }
+
+  private HttpServerResponse succeedingHttpServerResponse() {
+    HttpServerResponse httpServerResponse = mock(HttpServerResponse.class);
+    doAnswer(AdditionalAnswers.answerVoid(
+        (Handler<AsyncResult<Void>> handler) -> handler.handle(Future.succeededFuture())))
+        .when(httpServerResponse).end(any(Handler.class));
+    return httpServerResponse;
+  }
+
+  public void testFetch300(TestContext testContext) {
```

Comment:
What behaviour is this test intended to demonstrate? I'm not confident I can work it out from the code

Reply:
Can successfully process 300 records, more than RMB's cursor page size of 100.
Other APIs had regressions at 100:
https://issues.folio.org/browse/RMB-732
https://issues.folio.org/browse/MODINVSTOR-574
https://issues.folio.org/browse/MODINVSTOR-592

Reply:
> Can successfully process 300 records, more than RMB's cursor page size of 100.

By successfully process, are you referring to writing the rows from the database to the HTTP response?